### ISSUE TITLE
AddItem can fail the id assertion under certain conditions (Apple Silicon?)

### ISSUE THREAD

**Issue by juliusl** [imgui-rs/imgui-rs#771]
**Describe the bug**

If the window title is of a certain length, followed by an input widget (for the sake of this example InputText) w/ a certain length of label and hint, followed by a button w/ a label of a certain length, it can fail the assertion at line 9931 of Imgui.cpp.

**To Reproduce**
Steps to reproduce the behavior, or example code that triggers the bug.

This was tough to narrow down a repro for. It seems the characters used as well as lengths changes the overall result.

Will fail w/ 

`Assertion failed: (id != window->ID && "Cannot have an empty ID at the root of a window. If you need an empty label, use ## and read the FAQ about how the ID Stack works!"), function ItemAdd, file imgui.cpp, line 9183.`

```rs
imgui
            .window("aaaaaaa: aaa-aaaaaaaa-0.0.0") 
            .build(|| {
                let mut sc = String::new();
                imgui.input_text("aaaaaa aaaaaaaa##aaaaaa", &mut sc)
                    .hint("Type '*' to list all resources")
                    .enter_returns_true(true)
                    .build();
                imgui.button("aaaaaa##aaaaaaaa");
});
```

Will work (only difference is the hint is commented out)

```rs
imgui
            .window("aaaaaaa: aaa-aaaaaaaa-0.0.0") 
            .build(|| {
                let mut sc = String::new();
                imgui.input_text("aaaaaa aaaaaaaa##aaaaaa", &mut sc)
                    // .hint("Type '*' to list all resources")
                    .enter_returns_true(true)
                    .build();
                imgui.button("aaaaaa##aaaaaaaa");
});
```

**Expected behavior**

**Screenshots**

**Please describe your environment**
 - imgui-rs version [e.g. 0.4.0]: 0.11.0
 - Operating system version: Mac OS 14.2.1
- Apple M1 Max
- winit, wgpu, native

**Other information**
Add related issues, suggestions about how to fix, any other context here.

This doesn't seem to repro on Ubuntu 22.04 on x86-64 which is why I think it might be hardware related. It seems related to the scratch_txt buffer.

**Comment by salvatorebenedetto** [comment]
I believe this is due to #762 which has been fixed in cef617ba029bd5922043bc7bb3a4039cfa92357e but it is not included in `0.11.0`, so until they make a new release you can try and check out manually cef617ba029bd5922043bc7bb3a4039cfa92357e or later.

**Comment by juliusl** [comment]
@salvatorebenedetto  thanks, your explanation clears that up. Hoping to see 0.12.0 soon.

**Comment by sanbox-irl** [comment]
Yup, going to close as solved in v0.12.0!
